boltdesignsystem/bolt
# Update Popover to mirror it's internal state with the original no-JS / SSR state

## Jira
http://vjira2:8080/browse/BDS-2022

## Summary
Updates the Popover component to automatically update it's internal JS state if previously opened before the page's JS fires + remove any related URL hashes. 

![CleanShot 2020-03-12 at 13 28 49](https://user-images.githubusercontent.com/1617209/76548686-855a8d00-6465-11ea-96fd-a3e3813a2f4b.gif)

This PR also adds 3 new Nightwatch E2E integration tests for the Popover component to make sure that:
1. A popover that was previously opened before the page's JS kicks in will remain open, can be closed per usual, and have the hashed URL cleaned up.
2. A popover that was previously opened and _closed_ before the page's JS kicks in will remain closed and have the hashed URL cleaned up.
3. The URL cleanup logic doesn't interfere with any existing query strings

## How to test
- Review code updates
- Confirm new E2E integration tests working as expected
- Try out the new "state restoration" feature by visiting a URL [like this one](https://boltdesignsystem-i8ngo75i2.now.sh/pattern-lab/patterns/02-components-popover-25-popover-theme-variations/02-components-popover-25-popover-theme-variations.html#js-bolt-popover-theme-demo-light-3) that includes a hash that matches a particular Popover on the page and confirm the hash is removed + popover remains open.
